### DISCUSSION TITLE
API calls to relative `/api/` endpoint is not working inside `getServerSideProps`

### DISCUSSION THREAD

**Discussion by harrisrobin** [vercel/next.js#14604] [Help]
# Bug report

## Describe the bug

I'm calling a certain api (inside the /pages/api folder) in getServerSideProps and I'm getting an error i don't really understand. 

## To Reproduce

here's what my code looks like: 
```js
import zeitFetch from "@zeit/fetch"

const fetch = zeitFetch()

export const getServerSideProps: GetServerSideProps = async context => {
  const cookies = parseCookies(context)
  const { req } = context
  const { origin } = absoluteUrl(req, "localhost:3000")
  const token = cookies.token
  try {
    await fetch(`${origin}/api/favorites/list`, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
  } catch (error) {
    console.log({ error })
  }

  return { props: { cookies } }
}
```

This is suppose to make a call to the absolute url, which locally would be `http://localhost:3000/api/favourites/list` however I always get the following error: 

![image](https://user-images.githubusercontent.com/3880008/85903011-55948280-b7d3-11ea-937f-d72990bfdb84.png)

```
{ Error: queryA ENOTFOUND localhost
    at QueryReqWrap.onresolve [as oncomplete] (dns.js:196:19)
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND',
  syscall: 'queryA',
  hostname: 'localhost' }
```

## Expected behavior

I was hoping it would hit my api endpoint.

**Comment by transiient** [comment]
Replace `localhost:3000` with `127.0.0.1:3000`. `localhost` is an alias for this address - and it looks like, somewhere, this alias isn't actually working on your system. Could be an issue with your `hosts` file.
